### PR TITLE
Add Intel Extension flavor to SKLearn Containers

### PR DESCRIPTION
**Pull request by sramakintel**
This PR adds container support for Intel Extension for SKLearn alongside its stock version. 

This is tested on the linear regression sample on an Azure ML Instance. 

Notes:
During dockerbuild, `{{latest-image-tag}}` cannot be parsed using `docker build` command. Current solution in the PR is to use `sed` to make the variable substitution. The other possible solution is to hard code the latest version. Please suggest an alternative.

### CODE REVIEW

**Comment by sramakintel** [comment]
Hello team,

Please find my reply below


@microsoft-github-policy-service agree company="Intel Corporation"


Regards,
Srikanth

File: training/general/src/environments/sklearn-1.1-ubuntu20.04-py38-cpu/context/intel-extension/conda_dependencies.yaml [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+channels:
+- conda-forge
+- anaconda
+dependencies:
+- python=3.9
+- pip=21.3.1
+- pandas~=1.3.0
+- scipy~=1.7.0
+- numpy~=1.21.0
+- scikit-learn-intelex~={{latest-sklearnex-version}}
```
Comment (savitamittal1) [inline training/general/src/environments/sklearn-1.1-ubuntu20.04-py38-cpu/context/intel-extension/conda_dependencies.yaml:11]: does the env identifies latest-sklearnex-version?

Comment (savitamittal1) [inline training/general/src/environments/sklearn-1.1-ubuntu20.04-py38-cpu/context/intel-extension/conda_dependencies.yaml:11]: what is the size of intel extensions?

File: training/general/src/environments/sklearn-1.1-ubuntu20.04-py38-cpu/context/README.md [hunk 2]
```diff
@@ -0,0 +1,35 @@
+# SKLearn Docker Environment
```
Comment (savitamittal1) [inline training/general/src/environments/sklearn-1.1-ubuntu20.04-py38-cpu/context/README.md:1]: change folder name to be <framework>-<framework version>-<gpu driver>-<driver version>
@cassieesvelt is working on updating env name for all images and can provide the exact name for sklearn.

File: training/general/src/environments/sklearn-1.1-ubuntu20.04-py38-cpu/context/intel-extension/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,17 @@
+FROM mcr.microsoft.com/azureml/openmpi4.1.0-ubuntu20.04:{{latest-image-tag}}
```
Comment (savitamittal1) [inline training/general/src/environments/sklearn-1.1-ubuntu20.04-py38-cpu/context/intel-extension/Dockerfile:1]: Will you be updating in all other envs as well?

File: training/general/src/environments/sklearn-1.1-ubuntu20.04-py38-cpu/context/intel-extension/conda_dependencies.yaml (hunk 1 above)
Comment (sramakintel) [reply training/general/src/environments/sklearn-1.1-ubuntu20.04-py38-cpu/context/intel-extension/conda_dependencies.yaml:11]: The size of the intel extension docker image  is 2.68GB and that of the stock version is 2.89 GB

File: training/general/src/environments/sklearn-1.1-ubuntu20.04-py38-cpu/context/intel-extension/Dockerfile (hunk 3 above)
Comment (sramakintel) [reply training/general/src/environments/sklearn-1.1-ubuntu20.04-py38-cpu/context/intel-extension/Dockerfile]: @elesel This PR was work in progress. Was this closed by mistake? I would like to re-open it and work through the PR merge process.

Comment (elesel) [reply training/general/src/environments/sklearn-1.1-ubuntu20.04-py38-cpu/context/intel-extension/Dockerfile]: The PR was closed by GitHub as a side effect when I force pushed to all branches to remove some training data from git history. Would you please create a new PR from your existing branch?

Comment (sramakintel) [reply training/general/src/environments/sklearn-1.1-ubuntu20.04-py38-cpu/context/intel-extension/Dockerfile]: @elesel : I don't see a Re-open pull request option here. Does owner have the option to re-open?

Comment (elesel) [reply training/general/src/environments/sklearn-1.1-ubuntu20.04-py38-cpu/context/intel-extension/Dockerfile]: No, there's not an option to re-open this. Please create a new PR from the existing branch.

Comment (sramakintel) [reply training/general/src/environments/sklearn-1.1-ubuntu20.04-py38-cpu/context/intel-extension/Dockerfile]: i tried to create a new PR on my forked repo and pushed a small change to itn

I get this:
image

Comment (elesel) [reply training/general/src/environments/sklearn-1.1-ubuntu20.04-py38-cpu/context/intel-extension/Dockerfile]: I'm sorry--I didn't notice that you were working from a fork. At this point I'd recommend ensuring you have a copy of your changes and then re-forking azureml-assets and re-applying your changes to the new fork.

Comment (sramakintel) [reply training/general/src/environments/sklearn-1.1-ubuntu20.04-py38-cpu/context/intel-extension/Dockerfile]: added a new PR- https://github.com/Azure/azureml-assets/pull/392 after re-forking and re-applying changes. Please review when you get a chance. Thanks.

**Pull request closed without merging**